include/libtorrent/extensions.hpp

```diff
@@ -214,6 +219,10 @@ namespace libtorrent
 		// called when plugin is added to a session
 		virtual void added(session_handle) {}
 
+		// called after a plugin is added
+		// allows the plugin to register DHT requests it would like to handle
+		virtual void register_dht_extensions(dht_extensions_t&) {}
```

Comment:
Yes, it's a map of query strings to handlers. Do typedefs consisting entirely of standard/exported types need to exported? I didn't think they did.